internal/util/flowgraph/node.go

```diff
@@ -98,8 +103,17 @@ func (nodeCtx *nodeCtx) work() {
 				nodeCtx.collectInputMessages()
 				inputs = nodeCtx.inputMessages
 			}
-			n := nodeCtx.node
-			res = n.Operate(inputs)
+			if isCloseMsg(inputs) {
```

Comment:
a little bit confused here, how about:
```
if isCloseMsg(inputs) {
    res = inputs
    close(nodeCtx.closeCh)
    nodeCtx.node.Close()
} else {
    n := nodeCtx.node
    res = n.Operate(inputs)
}
```

Reply:
you can understand the input node decides whether the `operate` method is executed and the `res` decide whether the node should be closed.